pandas/core/frame.py

```diff
@@ -8548,6 +8462,140 @@ def isin(self, values) -> "DataFrame":
 ops.add_special_arithmetic_methods(DataFrame)
 
 
+@functools.singledispatch
+def create_block_manager(
+    data: Any,
+    df: DataFrame,
+    index: Optional[Axes],
+    columns: Optional[Axes],
+    dtype: Optional[Dtype],
+    copy: bool,
+) -> BlockManager:
+    """
+    Convert an object into a BlockManager. Used inside the DataFrame constructor
+    so if you want to provide a custom way to convert from your objec to a DataFrame
+    you can register a dispatch on this method.
+    """
+    # Base case is to try to cast to NumPy array
+    try:
+        arr = np.array(data, dtype=dtype, copy=copy)
+    except (ValueError, TypeError) as err:
+        exc = TypeError(
+            f"DataFrame constructor called with incompatible data and dtype: {err}"
+        )
+        raise exc from err
+
+    if arr.ndim == 0 and index is not None and columns is not None:
+        values = cast_scalar_to_array((len(index), len(columns)), data, dtype=dtype)
+        return init_ndarray(values, index, columns, dtype=values.dtype, copy=False)
+    else:
+        raise ValueError("DataFrame constructor not properly called!")
+
+
+@create_block_manager.register
+def _create_block_manager_none(data: None, *args, **kwargs):
+    return create_block_manager({}, *args, **kwargs)
+
+
+@create_block_manager.register
```

Comment:
not in love with this name

maybe create_dataframe ?

Reply:
Sure I can rename it to that. Might just be a bit confusing though since the return type is a block manager, not a dataframe.